LibVLCSharp.WPF/VideoView.cs

```diff
@@ -1,96 +1,119 @@
+            if (!IsDesignMode)
             {
-                var content = Content;
-                Content = null;
-                _foreground.Content = (UIElement)content;
+                var windowsFormsHost = Template.FindName(PART_PlayerHost, this) as FrameworkElement;
+                if (windowsFormsHost != null)
+                {
+                    ForegroundWindow = new ForegroundWindow(windowsFormsHost);
+                    ForegroundWindow.Content = ViewContent;
+                }
+
+                var hwnd = (Template.FindName(PART_PlayerView, this) as System.Windows.Forms.Panel)?.Handle;
+                if (hwnd != null)
+                {
+                    MediaPlayer.Hwnd = (IntPtr)hwnd;
+                }
             }
         }
 
-        void VLCResize()
+        protected override void OnContentChanged(object oldContent, object newContent)
         {
-            uint h = 0, w = 0;
+            base.OnContentChanged(oldContent, newContent);
 
-            if (MediaPlayer.Size(0, ref w, ref h))
+            if (IsDesignMode || IsUpdatingContent)
             {
-                var scalew = (float)_controlWidth / w;
-
-                var scaleh = (float)_controlHeight / h;
-
-                var scale = scalew < scaleh ? scalew : scaleh;
-                MediaPlayer.Scale = scale;
+                return;
             }
-        }
 
-        void OnSizeChanged(object sender, SizeChangedEventArgs e)
-        {
-            _controlWidth = e.NewSize.Width;
-            _controlHeight = e.NewSize.Height;
+            IsUpdatingContent = true;
+            try
+            {
+                Content = null;
+            }
+            finally
+            {
+                IsUpdatingContent = false;
+            }
 
-            if (MediaPlayer.IsPlaying)
+            ViewContent = newContent as UIElement;
+            if (ForegroundWindow != null)
             {
-                VLCResize();
+                ForegroundWindow.Content = ViewContent;
             }
         }
 
         public void Dispose()
         {
-            if (MediaPlayer.IsPlaying)
-                MediaPlayer.Stop();
-
-            SizeChanged -= OnSizeChanged;
-            Loaded -= VideoView_Loaded;
             Unloaded -= VideoView_Unloaded;
 
-            MediaPlayer.Hwnd = IntPtr.Zero;
-            MediaPlayer.Dispose();
-            LibVLC.Dispose();
-        }
+            var mediaPlayer = MediaPlayer;
+            if (mediaPlayer != null)
+            {
+                if (mediaPlayer.IsPlaying)
+                    mediaPlayer.Stop();
+                mediaPlayer.Hwnd = IntPtr.Zero;
+                mediaPlayer.Dispose();
+                MediaPlayer = null;
+            }
 
-        public MediaPlayer MediaPlayer { get; }
-        public LibVLC LibVLC { get; }
+            var libVLC = LibVLC;
+            if (libVLC != null)
+            {
+                libVLC.Dispose();
+                LibVLC = null;
```

Comment:
You're disposing `libVLC` and setting null to `LibVLC`. Is that on purpose? The small casing difference can lead to confusion.

Reply:
Yes, that's the purpose. I'm going to not use an intermediary variable, it's useless.
I added a finalizer that also call the Dispose method because in some cases (application shutdown for example), the Unloaded event is not called. So, in the other cases, the Dispose method will be called twice (in the Unloaded event and the finalizer).
I don't know if it is a good thing to call the Dispose method in the UnLoaded event because we can't remove the control from the visual tree without disposing the MediaPlayer.